kore/test/Test/Kore/Builtin/KEqual.hs

```diff
@@ -74,12 +75,14 @@ testBinary symb impl =
 test_KEqual :: [TestTree]
 test_KEqual =
     [ testCaseWithoutSMT "dotk equals dotk" $ do
-        let expect = Pattern.fromTermLike $ Test.Bool.asInternal True
+        let expect =
+                MultiOr.singleton . Pattern.fromTermLike $
+                    Test.Bool.asInternal True
             original = keqBool dotk dotk
         actual <- evaluate original
         assertEqual' "" expect actual
     , testCaseWithoutSMT "kseq(x, dotk) equals kseq(x, dotk)" $ do
-        let expect = Pattern.top
+        let expect = MultiOr.singleton Pattern.top
```

Comment:
Could possibly abstract out a:
```haskell
orTop = MultiOr.singleton Pattern.top
```
along with:
```haskell
orBottom = MultiOr.singleton Pattern.bottom
```
if you think it'd improve the repetition.

Reply:
These are already `Kore.Internal.OrPattern.top` and `bottom`.

Reply:
Actually `bottom` is implemented as `fromPatterns []`

Reply:
An `OrPattern`, by construction, will never contain `Pattern.bottom`. I think we should erase that TODO because I remember we tried that and there were some issues we couldn't address nicely.

Reply:
@ana-pantilie I checked [this](https://github.com/kframework/kore/blob/ba52e5fcacfbc1e887c34495568b811bcdf438e7/kore/test/Test/Kore/Builtin/Builtin.hs#L250) `patterns` on our unit tests.
Adding
```
        when (patterns == MultiOr.singleton Pattern.bottom) $
            error "Found singleton bottom OrPattern"
```
in `evaluate` breaks a lot of tests.

Reply:
@andreiburdusa [MultiOr.singleton](https://github.com/kframework/kore/blob/ba52e5fcacfbc1e887c34495568b811bcdf438e7/kore/src/Kore/Internal/MultiOr.hs#L163) of `Pattern.bottom` will be an empty `MultiOr`.